src/main/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessage.java

```diff
@@ -77,6 +115,59 @@ public Sha256Hash getSigHash() {
         return sigHash;
     }
 
+    public JsonNode getMessageToSign(int version) {
+        ObjectNode messageToSend = new ObjectMapper().createObjectNode();
+        messageToSend.put(TX.getFieldName(), getBtcTransactionSerialized());
+        messageToSend.put(INPUT.getFieldName(), inputIndex);
+        if (version == HSM_VERSION_5) {
+            if (hasWitness()) {
+                populateWithSegwitValues(messageToSend);
+            } else {
+                populateWithLegacyValues(messageToSend);
+            }
+        }
+        return messageToSend;
+    }
+
+    private JsonNode populateWithLegacyValues(ObjectNode messageToSend) {
+        messageToSend.put(SIGHASH_COMPUTATION_MODE.getFieldName(), SIGHASH_LEGACY_MODE);
+        return messageToSend;
```

Comment:
This `returns` but nothing is done with the returned value

Reply:
Remember to check intellij warnings, it highlights these things

Reply:
Solved.